### PR TITLE
Update docs for 2.0 development

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
readme & contribution docs

### Description
This PR updates the readme and contribution docs with information about the ongoing 2.0 development.
I cherry picked the restructuring of `CONTRIBUTING.md` from `master` and added a note about 2.0 afterwards.

### Steps
- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/master/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,106 +1,65 @@
...
-works best for you.
+[matrix.org-image]: https://img.shields.io/matrix/hedgedoc:matrix.org?logo=matrix&server_fqdn=matrix.org
 
-CodiMD can integrate with
+[matrix.org-url]: https://chat.hedgedoc.org
 
-* facebook, twitter, github, gitlab, dropbox, google, ldap, saml and [oauth2](docs/guides/auth/oauth.md) **for login**
-* imgur, s3, minio, azure **for image/attachment storage** (files can also be local!)
-* dropbox **for export and import**
+[matrix.org-dev-url]: https://matrix.to/#/#hedgedoc-dev:matrix.org?via=shivering-isles.com&via=matrix.org&via=kif.rocks
```
Comment (ErikMichelson) [inline README.md:41]: For later changes (and a shorter URL) just use: https://chat.hedgedoc.org/dev

Comment (davidmehren) [reply README.md:41]: Didn't know that exists, thanks xD

**Comment by mrdrogdrog** [comment]
Maybe you want to wait for https://github.com/hedgedoc/hedgedoc/pull/650 and/or include these changes?

**Comment by davidmehren** [comment]
@mrdrogdrog I'm not sure what I should include, #650 does not change the readme or contribution docs.
Of course we will need to forward-port the docs (and other stuff) to the `develop` branch.

**Comment by mrdrogdrog** [comment]
> we will need to forward-port the docs (and other stuff) to the develop branch.

That was what I meant. Good that you already got that covered :ok_hand:

**Review by InnayTool — APPROVED** [review]

**Review by mrdrogdrog — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,37 +1,81 @@
...
+
+If you set your `user.name` and `user.email` git configs, you can sign your commit automatically with `git commit -s`.
+You can also use git [aliases](https://git-scm.com/book/tr/v2/Git-Basics-Git-Aliases)
+like `git config --global alias.ci 'commit -s'`. Now you can commit with `git ci` and the commit will be signed.
+
+## Submitting a Pull Request
+
+1. Submit an issue describing your proposed change. We will try to respond to your issue promptly.
+2. Fork this repo, develop and test your code changes. Ensure you signed all your commits (see above for details).
+3. Submit a pull request against this repo's `master` branch.
```
Comment (DerMolly) [inline CONTRIBUTING.md:68]:
```suggestion
3. Submit a pull request against this repo's `develop` branch.
```

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (davidmehren) [reply CONTRIBUTING.md]: fixed, thx

**Review by DerMolly — APPROVED** [review]

**Pull request merged**